lib/WorseReflection/Core/Reflection/TypeResolver/MethodTypeResolver.php

```diff
@@ -52,7 +52,7 @@ private function resolveType(Type $type): Type
             return $type;
         }
 
-        return GenericHelper::resolveMethodType(
+        return StaticGenericHelper::resolveMethodType(
```

Comment:
the original "helper" conflicts with the new "runtime" resolution. the follow up ticket #1594 should remove it completely.